cms/djangoapps/contentstore/signals/handlers.py

```diff
@@ -51,18 +52,23 @@ def listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=
     registering proctored exams, building up credit requirements, and performing
     search indexing
     """
-
-    # first is to registered exams, the credit subsystem will assume that
-    # all proctored exams have already been registered, so we have to do that first
-    try:
-        register_special_exams(course_key)
-    # pylint: disable=broad-except
-    except Exception as exception:
-        log.exception(exception)
-
-    # then call into the credit subsystem (in /openedx/djangoapps/credit)
-    # to perform any 'on_publish' workflow
-    on_course_publish(course_key)
+    course_key_str = str(course_key)
```

Comment:
#NIT Why not this line, which is only used in the async case, into the `if is_enabled:` statement?